editors/code/src/test_explorer/run_or_debug.ts

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable no-console */
+import * as vscode from "vscode";
+import * as toolchain from "../toolchain";
+import { testController } from ".";
+import { spawn } from "child_process";
+import { assert } from "../util";
+import { createArgs, prepareEnv } from "../run";
+import { getRunnableByTestItem } from "./discover_and_update";
+import { TestControllerHelper } from "./TestControllerHelper";
+import { startDebugSession } from "../debug";
+import { raContext } from "../main";
+import { RustcOutputAnalyzer } from "./RustcOutputAnalyzer";
+
+export async function runHandler(
+    request: vscode.TestRunRequest,
+    token: vscode.CancellationToken
+) {
+    // TODO: Never run tests concurrently in client side.
+
+    const chosenItems = await getChosenTestItems(request);
+
+    if (!chosenItems) {
+        return;
+    }
+
+    const testRun = testController.createTestRun(request);
+
+    switch (request.profile?.kind) {
+        case vscode.TestRunProfileKind.Debug:
+            await debugChosenTestItems(chosenItems, token);
+            return;
+        case vscode.TestRunProfileKind.Run:
+            await runChosenTestItems(chosenItems, token, testRun);
+            return;
+        default:
+            await vscode.window.showErrorMessage("NOT SUPPORT YET");
```

Comment:
When might this occur?

Reply:
There is an option called "Coverage", but I do not know when it will be called.

Reply:
Seems the coverage is only called by program rather than user